src/SmartTransactionsController.ts

```diff
@@ -363,13 +345,47 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
       ethQuery = new EthQuery(networkClient.provider);
     }
 
-    this.#updateSmartTransaction(smartTransaction, {
+    this.#createOrUpdateSmartTransaction(smartTransaction, {
       chainId,
       ethQuery,
     });
   }
 
-  async #updateSmartTransaction(
+  #updateSmartTransaction(
+    smartTransaction: SmartTransaction,
+    {
+      chainId = this.config.chainId,
+    }: {
+      chainId: Hex;
+    },
+  ) {
+    const { smartTransactionsState } = this.state;
+    const { smartTransactions } = smartTransactionsState;
+    const currentSmartTransactions = smartTransactions[chainId] ?? [];
+    const currentIndex = currentSmartTransactions?.findIndex(
+      (stx) => stx.uuid === smartTransaction.uuid,
+    );
+    if (currentIndex === -1) {
+      return; // Smart transaction not found, don't update anything.
+    }
+    this.update({
+      smartTransactionsState: {
+        ...smartTransactionsState,
+        smartTransactions: {
+          ...smartTransactionsState.smartTransactions,
+          [chainId]: smartTransactionsState.smartTransactions[chainId].map(
+            (item, index) => {
+              return index === currentIndex
+                ? { ...item, ...smartTransaction }
```

Comment:
Just a heads-up: this method updates only the properties present in the provided `smartTransaction`. This merge-like behavior might be surprising for some users. Case in point, looking at line 593 in this same PR, if "merge" were the expected bahaviour, you could just write `{confirmed: true}`.

Reply:
We could only pass `{ confirmed: true }` on line 593, but the `#updateSmartTransaction` function expects the `SmartTransaction` type for the `smartTransaction` input param, which also requires `uuid`. I believe that for simplicity it's easier to just pass the whole `smartTransaction` to be updated in state.